coreblocks/func_blocks/fu/fpu/fpu_rounding_module.py

```diff
@@ -0,0 +1,257 @@
+from amaranth import *
+from amaranth.lib.wiring import Component, In, Out, Signature
+from transactron import TModule, Method, def_method
+from coreblocks.func_blocks.fu.fpu.fpu_common import (
+    RoundingModes,
+    FPUParams,
+    FPURoundingParams,
+)
+
+
+class FPURoundingSignature(Signature):
+    """FPU Rounding module signature
+
+    Parameters
+    ----------
+    fpu_params: FPUParams
+        FPU parameters
+    """
+
+    def __init__(self, *, fpu_params: FPUParams):
+        super().__init__(
+            {
+                "in_sign": In(1),
+                "in_sig": In(fpu_params.sig_width),
+                "in_exp": In(fpu_params.exp_width),
+                "rounding_mode": In(3),
+                "guard_bit": In(1),
+                "sticky_bit": In(1),
+                "in_errors": In(3),
+                "out_sign": In(1),
+                "out_sig": Out(fpu_params.sig_width),
+                "out_exp": Out(fpu_params.exp_width),
+                "out_error": Out(3),
+            }
+        )
+
+
+class FPURoudningMethodLayout:
+    """FPU Rounding module layouts for methods
+
+    Parameters
+    ----------
+    fpu_params: FPUParams
+        FPU parameters
+    """
+
+    def __init__(self, *, fpu_params: FPUParams):
+        self.rounding_in_layout = [
+            ("sign", 1),
+            ("sig", fpu_params.sig_width),
+            ("exp", fpu_params.exp_width),
+            ("guard_bit", 1),
+            ("sticky_bit", 1),
+            ("rounding_mode", 3),
```

Comment:
enums can be used as signal shapes
```suggestion
            ("rounding_mode", RoundingModes),
```

Reply:
Done